text/0000-relax-mixin-contract.md

```diff
@@ -0,0 +1,69 @@
+# Meta
+[meta]: #meta
+- Name: Relax Mixin Contract
+- Start Date: 2020-08-12
+- Author(s): @sclevine
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+Mixins are used to imply multiple contracts:
+
+1. A contract between a buildpack and a stack, where the buildpack may state that it can only be used with stack images that have certain dependencies.
+
+2. A contract between a stack run image and stack build image, where apps built on the build image must have a certain dependency present on the run image to start successfully on the run image.
+
+Currently, if a stack build image has the mixin `build:mypkg` and a stack run image has the mixin `run:mypkg`, a buildpack cannot require the mixin as `mypkg` (either statically in buildpack.toml, or dynamically via the stackpack mechanism described in #87). A stack author may choose to use this configuration because the same dependency is used in different, unrelated ways in both images.
```

Comment:
```suggestion
Currently, if a stack build image has the mixin `build:mypkg` and a stack run image has the mixin `run:mypkg`, a buildpack cannot require the mixin as `mypkg` (either statically in buildpack.toml, or dynamically via the stackpack mechanism described in [#87](https://github.com/buildpacks/rfcs/issues/87)). A stack author may choose to use this configuration because the same dependency is used in different, unrelated ways in both images.
```